src/libs/actions/Delegate.ts

```diff
@@ -0,0 +1,35 @@
+import Onyx from 'react-native-onyx';
+import * as API from '@libs/API';
+import {SIDE_EFFECT_REQUEST_COMMANDS} from '@libs/API/types';
+import * as NetworkStore from '@libs/Network/NetworkStore';
+import * as SequentialQueue from '@libs/Network/SequentialQueue';
+import {openApp} from './App';
+import updateSessionAuthTokens from './Session/updateSessionAuthTokens';
+
+function connect(email: string) {
+    // eslint-disable-next-line rulesdir/no-api-side-effects-method
+    API.makeRequestWithSideEffects(SIDE_EFFECT_REQUEST_COMMANDS.CONNECT_AS_DELEGATE, {to: email}, {})
+        .then((response) => {
+            if (!response?.restrictedToken || !response?.encryptedAuthToken) {
+                // TODO: Show an error message to the user
+                console.error('Error during connect: No authToken returned');
+                return;
+            }
+            return SequentialQueue.waitForIdle()
+                .then(() => Onyx.clear())
+                .then(() => {
+                    // Update authToken in Onyx and in our local variables so that API requests will use the new authToken
+                    updateSessionAuthTokens(response?.restrictedToken, response?.encryptedAuthToken);
+
+                    NetworkStore.setAuthToken(response?.restrictedToken ?? null);
+                    openApp();
```

Comment:
Not sure if there's anything we can do about this but when we load the new token we see the signin page for an instant. Can we do a full page loading instead, or some other solution?

Reply:
I think we'll have to preserve some onyx keys and not clear them

Reply:
preserving the key session does the job. resolved 👍